tracer/src/Datadog.Tracer.Native/debugger_rejit_preprocessor.cpp

```diff
@@ -238,14 +238,14 @@ const std::unique_ptr<RejitHandlerModuleMethod>
 DebuggerRejitPreprocessor::CreateMethod(const mdMethodDef methodDef, RejitHandlerModule* module,
                                         const FunctionInfo& functionInfo, const MethodProbeDefinition& methodProbe)
 {
-    return std::make_unique<DebuggerRejitHandlerModuleMethod>(methodDef, module, functionInfo);
+    return std::make_unique<DebuggerRejitHandlerModuleMethod>(methodDef, module, functionInfo, std::make_unique<DebuggerMethodRewriter>(m_corProfiler));
```

Comment:
We could've keep the rewriters as singletons by passing `CorProfiler` to the `Rewrite(...)` methods and then keep everything as sit was (I am guessing?). But I like this refactoring since it also cleaned various method signatures from `CorProfiler`.
I have only one concern/question, is the `CorProfiler` instance that triggers the preprocessor (and the one you are passing to `ModuleMethod`) is the one that needs to handle the rewriting?

Reply:
@GreenMatan In the P/Invoke, we retrieve the instance of CorProfilerInfo, so we might be using the wrong one (this PR does not intend to support mixed runtimes). However, each instance of CorProfilerInfo has its own ReJIT Preprocessor, so we shouldn't end up in a situation where we use one CorProfilerInfo for some of the processing and another for the rest (= we either use the right or the wrong instance for the *whole* processing, thus avoiding the crash).

So to answer your question, no we have no guarantee that we're using the right CorProfilerInfo instance, only that we're consistently using the same one.